packages/terra-date-picker/src/DateUtil.jsx

```diff
@@ -142,6 +143,103 @@ class DateUtil {
     return undefined;
   }
 
+  /**
+   * Gets the date format order from moment's long date format.
+   * @param {string} momentDateFormat - Moment's long date format.
+   * @return {string} - The date format order. Possible Values (MDY, DMY, YMD)
+   */
+  static getDateFormatOrder(momentDateFormat) {
+    let dateFormatOrder;
+    if (momentDateFormat === 'MM/DD/YYYY') {
+      dateFormatOrder = DateUtil.dateOrder.MDY;
+    } else if (momentDateFormat === 'DD/MM/YYYY') {
+      dateFormatOrder = DateUtil.dateOrder.DMY;
+    } else if (momentDateFormat === 'YYYY/MM/DD') {
+      dateFormatOrder = DateUtil.dateOrder.YMD;
+    }
+
+    return dateFormatOrder;
+  }
+
+  /**
+   * Returns an object consisting of date input values based on the date format.
+   * @param {string} dateOrder - String containing the date input order. Possible values (MDY, DMY, YMD)
+   * @param {string} value - The date/placeholder string.
+   * @param {string} separator - The date separator.
+   */
+  static getDateInputValues(dateOrder, value, separator) {
+    if (!value) {
+      return false;
+    }
+
+    let day;
+    let month;
+    let year;
+    const dateInputParts = value.split(separator);
+
+    switch (dateOrder) {
+      case DateUtil.dateOrder.DMY:
+        [day, month, year] = dateInputParts;
+        break;
+      case DateUtil.dateOrder.MDY:
+        [month, day, year] = dateInputParts;
+        break;
+      case DateUtil.dateOrder.YMD:
+        [year, month, day] = dateInputParts;
+        break;
+      default:
+        [year, day, month] = dateInputParts;
+    }
+
+    return { day, month, year };
+  }
+
+  /**
+   * Returns component layout based on locale format.
+   * @param {string} dateOrder - The date input order based on the locale date format.
+   * @param {object} separator - The separator component with the locale appropriate separator.
+   * @param {object} day - The date day component.
+   * @param {object} month - The date month component.
+   * @param {object} year - The date year component.
+   * @return {object} - The object containing the date input components in the order as per the locale date format.
+   */
+  static getInputLayout(dateOrder, separator, day, month, year) {
```

Comment:
The date format in the `nl` locale should be [DD-MM-YYYY](https://github.com/moment/moment/blob/develop/locale/nl.js#L69) but the input layout appears as YYYY-MM-DD. Perhaps, the `dateOrder` argument passed here is incorrect for the `nl` locale.

The same is happening for the `de` locale where the format should be [DD.MM.YYYY](https://github.com/moment/moment/blob/develop/locale/de.js#L48) but the input layout is YYYY.MM.DD

**nl**
![Screen Shot 2020-10-23 at 4 41 15 PM](https://user-images.githubusercontent.com/22618655/97056455-ad0ee900-154e-11eb-9808-f5027c2656ed.png)

**de**
![Screen Shot 2020-10-23 at 4 40 59 PM](https://user-images.githubusercontent.com/22618655/97056473-b5672400-154e-11eb-8c88-cd58fe4b3596.png)

Reply:
Addressed here : https://github.com/cerner/terra-framework/pull/1208/commits/ef6e9b4856db01f837879b86710bbe32a85e6a66